test/tests/vectorpostprocessors/execute_on_final/execute_on_final.i

```diff
@@ -0,0 +1,41 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 1
+[]
+
+[Problem]
+  solve = false
+[]
+
+[Executioner]
+  type = Transient
+  dt = 1
+  num_steps = 1
+[]
+
+[Functions]
+  [func]
+    type = ConstantFunction
+    value = 5
+  []
+[]
+
+[VectorPostprocessors]
+  [vpp]
+    type = LineFunctionSampler
+    functions = func
+    start_point = '0 0 0'
+    end_point = '1 0 0'
+    num_points = 4
+    sort_by = x
+    execute_on = final
+  []
+[]
+
+[Outputs]
+  [csv]
+    type = CSV
+    show = 'vpp'
+    execute_on = final
```

Comment:
I don't think this is a very good test because there's only one step and the answer would be the same with just timestep_end. A really good test would actually give a different answer on final when compared to just timestep_end, however, that might be pretty tricky to come up with. Perhaps you could have an AuxKernel or UserObject that also only executes on final so that you could see the output change on final? At the very least perhaps add another step so that you can verify that you get only one output.

Reply:
The purpose of the test was to demonstrate that VPP's `execute_on=FINAL` actually executed *before* output's `execute_on=FINAL`; if this is not true, then there are no points in the output file.

Reply:
oh you are right, I missed the fact that the VPP was executing on final...

Reply:
The new test includes both a PP and an Output that executes on final and one that does not.